.github/workflows/ca-clone-sequential-test.yml

```diff
@@ -1429,6 +1429,774 @@ jobs:
 
           diff expected actual
 
+      ####################################################################################################
+      # Switch cert request ID generator to newLegacy and verify if serials
+      # have gaps when range is updated
+      #
+      # It should work like the legacy but with correct range.
+      - name: Switch primary to newLegacy
+        run: |
+          # Update CS.cfg
+          docker exec primaryds ldapsearch \
+              -H ldap://primaryds.example.com:3389 \
+              -D "cn=Directory Manager" \
+              -w Secret.123 \
+              -b ou=certificateRepository,ou=ranges,dc=ca,dc=pki,dc=example,dc=com \
+              -s one \
+              -o ldif_wrap=no \
+              -LLL '(&(host=primary.example.com)(SecurePort=8443))' beginRange | tee output
+
+          # Since there is a range the value are retrieved from the range. In general, id range is not
+          # present the value from CS.cfg are used.
+          BEGIN_SERIAL=0x$(cat output | sed -n 's/beginRange: \(.*\)/\1/p' | sort -n | tail -1)
+          END_SERIAL=$(printf "0x%x\n" $(( BEGIN_SERIAL + 0x12 -1 )) )
+
+          docker exec primary pki-server ca-config-set dbs.beginSerialNumber $BEGIN_SERIAL
+          docker exec primary pki-server ca-config-set dbs.endSerialNumber $END_SERIAL
```

Comment:
I think these operations should be identical to `Repository.switchToNextRange()`, but the problem is the `switchToNextRange()` uses `mNextMinSerialNo` and `mNextMaxSerialNo` variables instead of `beginRange` and `endRange` attributes. I think they are supposed to be equivalent but computed separately so they might become out of sync because of the bug.

Now the question is should we use the next range defined in `mNextMinSerialNo` and `mNextMaxSerialNo` or in `beginRange` and `endRange`?

Reply:
Actually we are not switching range so the operation cannot be identical. When the generator is updated to the new version the range could still have space and it has to be used. We have to add a check and eventually update the next range configuration.

Reply:
I suppose you're converting this code into a tool? That would make it easier to implement the logic and reuse it in other cases.

Reply:
Could you also describe how the migration process works in this page? Thanks!
https://github.com/dogtagpki/pki/wiki/Sequential-Serial-Numbers-v2

Reply:
Yes, I have update the code but it is not yet complete. Beside the KRA update (they are the same so the update will work for both) which still has to be included, I have a couple of doubts I am still testing.

Additionally, since the range are switching to decimal as the next range (and the certificate serials) should we update the existing ranges which are already completed to be consistent?

Reply:
The migration works also for the RSNv3 but I have not modified the tests. We can do later.

Reply:
I have update the page with the steps performed by the update script. If it is OK then we could add an example of before and after the update from the `ca-sequential` action.